src/internal/easee/client.go

```diff
@@ -0,0 +1,387 @@
+package easee
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/thoas/go-funk"
+
+	"github.com/futurehomeno/edge-easee-adapter/internal/config"
+)
+
+const (
+	loginURI                  = "/api/accounts/token"
+	tokenRefreshURI           = "/api/accounts/refresh_token"
+	chargersURI               = "/api/chargers"
+	healthURI                 = "/health"
+	chargerControlURITemplate = "/api/chargers/%s/commands/%s"
+	chargerStateURITemplate   = "/api/chargers/%s/state"
+	cableLockURITemplate      = "/api/chargers/%s/commands/lock_state"
+
+	startChargingCommand = "resume_charging"
+	stopChargingCommand  = "pause_charging"
+
+	authorizationHeader = "Authorization"
+	contentTypeHeader   = "Content-Type"
+
+	jsonContentType = "application/*+json"
+)
+
+type Client interface {
```

Comment:
Just to confirm: auth flow is password based, user gives login and password, and then we login with it and store only refresh token and access token, right? How long the refresh token is valid?

Reply:
1. That's right.
2. Both access and refresh token are valid for the same amount of time - when refreshing tokens, the app receives a new pair of those.